lib/cache/adapters/memory_adapters/queue_adapter.rb

```diff
@@ -4,24 +4,32 @@ module Adapters
       module MemoryAdapters
         # Memory adapter implementation, which stores everything inside queue
         class QueueAdapter
-          def initialize(size)
-            @size = size
+          def initialize(max_size)
+            @max_size = max_size
             @queue = Queue.new
+            @current_size = Concurrent::AtomicFixnum.new(0)
           end
 
           # Adds data to queue in non-blocking mode
           def add_to_queue(data)
+            fail ThreadError if @current_size.value >= @max_size
```

Comment:
queue_adapter.rb uses it and does not catch `ThreadError`.

Is that an outdated code?
```ruby
          # Adds data to queue in non-blocking mode
          def add_to_queue(data)
            @queue.push(data)
          end
```